src/http-requests.ts

```diff
@@ -143,35 +143,36 @@ class HttpRequests {
     }
 
     const headers = { ...this.headers, ...config.headers };
+    const responsePromise = this.fetchWithTimeout(
+      constructURL.toString(),
+      {
+        ...config,
+        ...this.requestConfig,
+        method,
+        body,
+        headers,
+      },
+      this.requestTimeout,
+    );
 
-    try {
-      const result = this.fetchWithTimeout(
-        constructURL.toString(),
-        {
-          ...config,
-          ...this.requestConfig,
-          method,
-          body,
-          headers,
-        },
-        this.requestTimeout,
-      );
-
-      // When using a custom HTTP client, the response is returned to allow the user to parse/handle it as they see fit
-      if (this.httpClient) {
-        return await result;
-      }
+    const response = await responsePromise.catch((error: unknown) => {
+      throw new MeiliSearchRequestError(constructURL.toString(), error);
+    });
+
+    // When using a custom HTTP client, the response is returned to allow the user to parse/handle it as they see fit
+    if (this.httpClient !== undefined) {
+      return response;
+    }
 
-      const response = await result.then((res: any) =>
-        httpResponseErrorHandler(res),
-      );
-      const parsedBody = await response.json().catch(() => undefined);
+    const responseBody = await response.text();
+    const parsedResponse =
+      responseBody === '' ? undefined : JSON.parse(responseBody);
```

Comment:
Previously if `response.json()` failed, we swallowed the error. There are cases when the response body is empty, and is parsed to an empty string initially. That is an invalid JSON. This case was handled by simply swallowing the error, now it's handled explicitly, and if for whatever reason Meilisearch returns something non-JSON-valid other than an empty body, it will err, which is what we want, as that's entirely unexpected and should propagate.